sympy/integrals/tests/test_laplace.py

```diff
@@ -214,16 +224,18 @@ def test_laplace_transform():
     assert L - (s*cos(3) - sin(3))/(s**2 + 1) == 0
     # Error functions (laplace7.pdf)
     assert LT(erf(a*t), t, s) == (exp(s**2/(4*a**2))*erfc(s/(2*a))/s, 0, True)
-    assert LT(erf(sqrt(a*t)), t, s) == (sqrt(a)/(s*sqrt(a + s)), 0, True)
+    assert LT(erf(sqrt(a*t)), t, s) == (1/(s*sqrt(1 + s/a)), 0, True)
```

Comment:
In this case we have:
```python
In [25]: a, s = symbols('a, s')

In [26]: e1 = sqrt(a)/(s*sqrt(a + s)) # old

In [27]: e2 = 1/(s*sqrt(1 + s/a)) # new

In [28]: e1
Out[28]: 
    √a     
───────────
    _______
s⋅╲╱ a + s 

In [29]: e2
Out[29]: 
      1      
─────────────
      _______
     ╱     s 
s⋅  ╱  1 + ─ 
  ╲╱       a 

In [30]: e1.subs(a, -1)
Out[30]: 
     ⅈ     
───────────
    _______
s⋅╲╱ s - 1 

In [31]: e2.subs(a, -1)
Out[31]: 
     1     
───────────
    _______
s⋅╲╱ 1 - s 
```
These are not equal for `s > 1`:
```python
In [32]: e1.subs(a, -1).subs(s, 2)
Out[32]: 
ⅈ
─
2

In [33]: e2.subs(a, -1).subs(s, 2)
Out[33]: 
-ⅈ 
───
 2 
```

Reply:
Hmm, unfortunately Bateman54 gives the first formula, unless I made a stupid mistake. Is there a difference between $\sqrt{a}$ and $a^{1/2}$?
I hope I'm not wrong that $a^{1/2}t^{1/2}=(at)^{1/2}$ if t is positive and real...

Reply:
`sqrt(a)` and `a^(1/2)` are the same thing.

It is also true that if `t` is positive and real then `sqrt(a*t) = sqrt(a)*sqrt(t)`

For `a = -1` Wolfram alpha gives a Piecewise for this case:
https://www.wolframalpha.com/input?i=LT%28erf%28sqrt%28-t%29%29%2C+t%2C+s%29

It is basically `Piecewise((e2, s<1), (e1, s>1)`. I think that if `a = -1` then the two formulae are equivalent for `s < 1` though so it should be possible to have `e1` at least for real `s`.

Is `e1` what you mean by the first formula? If so and if that is the formula in Bateman then how have we eneded with `e2`? Maybe a `sqrt` is being computed in the wrong place?

The formula here looks like `e1`:
https://github.com/sympy/sympy/blob/b354658bfd7b863ee59897321d9645efbb9d1f57/sympy/integrals/laplace.py#L487-L488
Is there something higher up that extracts the `a`?

Reply:
Ah, I think I have it. In the test cases, `a` is defined as positive. With your new code,

```
a = symbols('a', real=True)
alpha = symbols('alpha', positive=True)
t = symbols('t', real=True)
print(sqrt(a*t))
print(sqrt(alpha*t))
```

immediately gives

```
sqrt(a*t)
sqrt(alpha)*sqrt(t)
```

before the LT can do anything, so my simple rule does not match anymore and the result is in the end achieved by integration. Something similar happens in all of the cases you cite.

For these test cases, the restriction to positive `a` in the tests was convenient such that not too many variables were needed, but it was not necessary. Now it bites us.

Howerver, we don't have a mathematical problem, since what happens here is correct for strictly positive `a`.

So I suggest the following:

Add
```
    ar, br, cr = symbols('ar, br, cr', real=True)
```
to the tests, and in all cases where an input like `sqrt(a*t)` fails, replace it by `sqrt(ar*t)`, and then it must give the same result as before your change to `as_base_exp`.

After you merge this, I will debug my pattern matching code to check why my simple rules cannot match the expression `erf(sqrt(alpha)*sqrt(t))` and instead defer them to integration.

Reply:
Okay, so is it the case that with or without the change here in this PR the Laplace code does not match and the rule is not used?

Does that mean that actually the changed output results from the integration code being affected by the change to `as_base_exp()` rather than it affecting anything in the Laplace-specific code?

Reply:
It would be good to use `ar` in the tests if `ar` has the real assumption set.

Reply:
And `ap` if the positive assumption is set.

Reply:
> Okay, so is it the case that with or without the change here in this PR the Laplace code does not match and the rule is not used?
> 
> Does that mean that actually the changed output results from the integration code being affected by the change to `as_base_exp()` rather than it affecting anything in the Laplace-specific code?

What happens with your change is that for positive `a` and in this PR, the rule is not used. It is used for real `a`. In sympy main it is used for both cases.

What happens in this PR is that for positive `a`, if you write `f=sqrt(a*t)`, the variable `f` will then contain `sqrt(a)*sqrt(t)`. Before this PR, the `sqrt` function did not do this.

Reply:
> It would be good to use `ar` in the tests if `ar` has the real assumption set.

True, so you could now, for this PR, implement the `ar`, and I will make a second update later to make the test code more readable by using `ap` only where it is really needed.